File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/EDIPartyName.cs
using System;
using System.Runtime.Serialization;

//	EDIPartyName ::= SEQUENCE {
//		nameAssigner [0] IMPLICIT DirectoryString OPTIONAL,
//		partyName    [1] IMPLICIT DirectoryString 
//	}

namespace Aladdin.ASN1.ISO.PKIX
{
	[Serializable]
	public class EDIPartyName : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ChoiceCreator<DirectoryString>().Factory(), Cast.O, Tag.Context(0)), 
			new ObjectInfo(new ChoiceCreator<DirectoryString>().Factory(), Cast.N, Tag.Context(1)), 
		}; 
		// конструктор при сериализации
        protected EDIPartyName(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public EDIPartyName(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public EDIPartyName(OctetString nameAssigner, OctetString partyName) : 
			base(info, nameAssigner, partyName) {}

		public OctetString NameAssigner	{ get { return (OctetString)this[0]; } } 
		public OctetString PartyName	{ get { return (OctetString)this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/EDIPartyName.java
package aladdin.asn1.iso.pkix;
import aladdin.asn1.*; 
import java.io.*;

//	EDIPartyName ::= SEQUENCE {
//		nameAssigner [0] IMPLICIT DirectoryString OPTIONAL,
//		partyName    [1] IMPLICIT DirectoryString 
//	}

public final class EDIPartyName extends Sequence<OctetString>
{
    private static final long serialVersionUID = 1627598266681309836L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ChoiceCreator(DirectoryString.class).factory(), Cast.O, Tag.context(0)), 
		new ObjectInfo(new ChoiceCreator(DirectoryString.class).factory(), Cast.N, Tag.context(1)), 
	}; 
	// конструктор при раскодировании
	public EDIPartyName(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public EDIPartyName(OctetString nameAssigner, OctetString partyName) 
	{
		super(info, nameAssigner, partyName); 
	}
	public final OctetString nameAssigner() { return get(0); } 	
	public final OctetString partyName   () { return get(1); }
}
